devicetypes/tonesto7/nest-thermostat.src/nest-thermostat.groovy

```diff
@@ -391,40 +395,50 @@ def tempUnitEvent(unit) {
 	def tmpUnit = device.currentState("temperatureUnit")?.value
 	if(!tmpUnit.equals(unit)) {   
     	log.debug("UPDATED | Temperature Unit: (${unit}) | Original State: (${tmpUnit})")
-        sendEvent(name:'temperatureUnit', value: unit, descriptionText: "Temperature Unit is now: '${unit}'", displayed: false, isStateChange: true)
+        sendEvent(name:'temperatureUnit', value: unit, descriptionText: "Temperature Unit is now: '${unit}'", displayed: true, isStateChange: true)
         state?.tempUnit = unit
     } else { Logger("Temperature Unit: (${unit}) | Original State: (${tmpUnit})") }
 }
 
-def targetTempEvent(targetTemp) {
-	def temp = device.currentState("targetTemperature")?.value.toString()
-	if(!temp.equals(targetTemp.toString())) {
-		log.debug("UPDATED | Target Temperature is (${targetTemp}) | Original Temp: (${temp})")
-		sendEvent(name:'targetTemperature', value: targetTemp, descriptionText: "Target Temperature is ${targetTemp}", displayed: true, isStateChange: true)
-	} else { Logger("Target Temperature is (${targetTemp}) | Original Temp: (${temp})") }
+def thermostatSetpointEvent(Double targetTemp) {
+    def temp = device.currentState("thermostatSetpoint")?.value.toString()
+    def rtargetTemp = wantMetric() ? targetTemp.round(1) : targetTemp.round(0).toInteger()
+    if(!temp.equals(rtargetTemp.toString())) {
+        log.debug("UPDATED | thermostatSetPoint Temperature is (${rtargetTemp}) | Original Temp: (${temp})")
+        sendEvent(name:'thermostatSetpoint', value: wantMetric() ? rtargetTemp.round(1) : targetTemp.round(0).toInteger(), unit: state?.tempUnit, descriptionText: "thermostatSetpoint Temperature is ${targetTemp}", displayed: false, isStateChange: true)
+    } else { Logger("thermostatSetpoint is (${targetTemp}) | Original Temp: (${temp})") }
 }
 
-def temperatureEvent(tempVal) {
+def temperatureEvent(Double tempVal) {
 	def temp = device.currentState("temperature")?.value.toString()
-	if(!temp.equals(tempVal.toString())) {
+    def rtempVal = wantMetric() ? tempVal.round(1) : tempVal.round(0).toInteger()
+    if(!temp.equals(rtempVal.toString())) {
         log.debug("UPDATED | Temperature is (${tempVal}) | Original Temp: (${temp})")
-    	sendEvent(name:'temperature', value: tempVal,  descriptionText: "Ambient Temperature is ${tempVal}" , displayed: true, isStateChange: true)
+    	sendEvent(name:'temperature', value: wantMetric() ? rtempVal : tempVal.round(0).toInteger(),  unit: state?.tempUnit, descriptionText: "Ambient Temperature is ${tempVal}" , displayed: false, isStateChange: true)
```

Comment:
I think you are correct (with proper spelling).

On Mar 23, 2016, at 9:18 AM, Tonesto7 <notifications@github.com<mailto:notifications@github.com>> wrote:


In devicetypes/tonesto7/nest-thermostat.src/nest-thermostat.groovy<https://github.com/tonesto7/nest-manager/pull/34#discussion_r57156797>:

>          log.debug("UPDATED | Temperature is (${tempVal}) | Original Temp: (${temp})")
> -     sendEvent(name:'temperature', value: tempVal,  descriptionText: "Ambient Temperature is ${tempVal}" , displayed: true, isStateChange: true)
> +     sendEvent(name:'temperature', value: wantMetric() ? rtempVal : tempVal.round(0).toInteger(),  unit: state?.tempUnit, descriptionText: "Ambient Temperature is ${tempVal}" , displayed: false, isStateChange: true)


Using the value: wantMetric() ? rTempVal : is kind of redundant right? Couldn't I just call value: rTempVal

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/tonesto7/nest-manager/pull/34/files/71738294e6f15465e5de7ddacf369c0b6f73c063#r57156797>